### PR TITLE
OAK-10656: MongoDocumentStore: keep metrics about document size related exceptions

### PR DESCRIPTION
**Pull request by reschke**
...we can later expand this to cover different exceptions.

### CODE REVIEW

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentStore.java [hunk 1]
```diff
@@ -2317,10 +2320,19 @@ private <T extends Document> DocumentStoreException handleException(Throwable ex
                 invalidateCache(collection, id);
             }
         }
+
+        if (ex instanceof MongoWriteException) {
```
Comment (stefan-egli) [inline oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentStore.java:2324]: Wondering if `MongoWriteException` is indeed the one we like to count.

Looking at the [MongoServerException](https://mongodb.github.io/mongo-java-driver/3.12/javadoc/com/mongodb/MongoServerException.html) it looks like there are bulk siblings that we might also be interested : `BulkWriteException`, `MongoBulkWriteException` ?

If that's the case, then I'm wondering if we should count `MongoServerException` instead - but arguably that's less specific than the current `MongoWriteException` ...

Comment (reschke) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentStore.java:2324]: I believe I checked, but will double-check again.

At the end of the day, we may want to count multiple types of exceptions, but I wanted to keep this one focused.

**Review by stefan-egli — APPROVED** [review]

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentStore.java (hunk 1 above)
Comment (stefan-egli) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentStore.java:2324]: +1 to go ahead with this one and then consider the bulk ones later

**Review by mbaedke — APPROVED** [review]

File: oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentStore.java (hunk 1 above)
Comment (reschke) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentStore.java:2324]: I verified that this is the right one; see https://issues.apache.org/jira/browse/OAK-10642?focusedCommentId=17819206&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17819206

Comment (reschke) [reply oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentStore.java:2324]: So proposal is to go ahead with this and expand this later on when we have more time.

**Pull request merged**